### ISSUE TITLE
GitHub Actions Build Failure

### ISSUE THREAD

**Issue by MikuAuahDark** [kcat/openal-soft#498]
Hello,

As part of my initiative of getting LOVE to compile in Windows 10 ARM64 (which is why I suddenly asking for ARM64 support in #494), looks like there's some change that caused OpenAL-soft fails to compile in latest GitHub Actions Windows runner. (bit related to #449)

I try to (ab)use GitHub Actions to create 15 build matrix and my conclusion is there's some change between 1.20.1 and 1.21.0 that breaks build in GitHub Actions, as can be seen in this test repository https://github.com/MikuAuahDark/test-oals-actions-build

Personally I can compile 1.21.0 (and even master branch) fine, however my automated GitHub Actions workflow can't generate artifacts.

Is there something I can help to get this fixed? If you need write access to that repo please let me know your email that's used for GitHub.  
Thanks.

**Comment by kcat** [comment]
As best as I am able to find ([here](https://github.com/microsoft/vcpkg/pull/12318#issuecomment-655782037)), it seems to be an issue relating to a Windows SDK being provided by the environment, and OpenAL Soft looking for and finding a different Windows SDK version, leading to conflicts between various headers. A quick fix may be to pass `-DCMAKE_DISABLE_FIND_PACKAGE_WindowsSDK=ON` when configuring OpenAL Soft. A more appropriate fix would likely be to find out if a Windows SDK is already supplied by the environment and skip trying to find one, though I don't know which of the multitude of VS-related CMake variables dictate that.

**Comment by MikuAuahDark** [comment]
I also tried to print all the environment variables of GitHub actions runner (same repo, see above) but I can't find any suspicious environment variable set (please double check to make sure). One thing I noticed is when I build locally, I see `Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19042.` while in GitHub Actions runner, I don't see that (or similar) line.

**Comment by kcat** [comment]
I'm not sure if it's a specific environment variable or not. It could be a registry setting, or a particular file somewhere. There could be some environment variables set later on before the initial cmake run. I do notice comparing to [an AppVeyor build](https://ci.appveyor.com/project/ChrisRobinson/openal-soft/build/job/e2l48lygs8297ar6), `DXSDK_DIR` doesn't seem to be set in GitHub Actions.

But I'm more interested in [any of the CMake variables](https://cmake.org/cmake/help/latest/search.html?q=CMAKE_VS_*) that may be getting set to indicate an SDK is already prepared/recognized (and is unset when it needs to be searched).

**Comment by MikuAuahDark** [comment]
Hello.

I try to add "Dump `CMAKE_VS_*` Variables" to the GitHub Actions step and here are the variables defined for GitHub runners
```
-- CMAKE_VS_DEVENV_COMMAND=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/devenv.com
-- CMAKE_VS_INTEL_Fortran_PROJECT_VERSION=11.0
-- CMAKE_VS_MSBUILD_COMMAND=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/MSBuild/Current/Bin/MSBuild.exe
-- CMAKE_VS_PLATFORM_NAME=x64
-- CMAKE_VS_PLATFORM_NAME_DEFAULT=x64
-- CMAKE_VS_PLATFORM_TOOLSET=v142
-- CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE=x64
-- CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION=10.0.17763.0
```
In contrast, this is my local build `CMAKE_VS_*` variables:
```
-- CMAKE_VS_DEVENV_COMMAND=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/devenv.com
-- CMAKE_VS_INTEL_Fortran_PROJECT_VERSION=11.0
-- CMAKE_VS_MSBUILD_COMMAND=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin/MSBuild.exe
-- CMAKE_VS_PLATFORM_NAME=x64
-- CMAKE_VS_PLATFORM_NAME_DEFAULT=x64
-- CMAKE_VS_PLATFORM_TOOLSET=v142
-- CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE=x64
-- CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION=10.0.19041.0
```

There's only 1 difference, the `CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION`. In GitHub runners, it's set to `10.0.17763.0` because GitHub runners are running Windows 10 1809 (Windows Server 2019), whilst my laptop is running Windows 10 20H2 (10.0.19042.610) and that variable is set to `10.0.19041.0` because that's highest SDK version available in my laptop.

I hope we can resolve this problem with information I provided above.

**Comment by MikuAuahDark** [comment]
Alright so I decided to do `git bisect` locally from `openal-soft-1.20.1` to `openal-soft-1.21.0` and pushing each commits to GitHub Actions, looks like e3f6afafdeab3c5ec8e28c3f9b6b26fbcf5f37bf is the culprit.

**Comment by kcat** [comment]
I'm really interested in why this doesn't seem to be a more wide-spread problem. The `FindWindowsSDK.cmake` module is used plenty elsewhere (something like it seems to be necessary on newer Windows/VS versions, since building fails if an SDK isn't searched for), yet OpenAL Soft seems to be the only thing having a problem with it, but then only in certain build environments. What's going on here.

**Comment by MikuAuahDark** [comment]
I think I have an idea how to replicate it locally.

1. Install Windows SDK `10.0.17763.0` **and** `10.0.19041.0`

2. Configure the `master` branch with `cmake -Bbuild -H. -DCMAKE_SYSTEM_VERSION=10.0.17763.0`

3. Try to build.

I haven't able to test this at the moment because I have datacap and I can't use that excessively.

**Comment by kcat** [comment]
Taking a look through everything, it seems FindWindowsSDK is only used to find `winmm.lib`, and maybe `dsound.h` and `dsound.lib`. And even then it should only affect the include paths when `DXSDK_DIR` doesn't have `dsound.h`. That would partly explain why it only affects certain build environments since GitHub Actions gets DSound through the Windows SDK, instead of the DX SDK like AppVeyor does.

It might be enough to modify `cmake/FindDSound.cmake` and remove `${WINSDK_INCLUDE_DIRS}` from the `find_path` call. It could still be needed for a system that doesn't find it in either the compiler's predefined paths or the DX SDK path... though I don't know if such an occurrence is possible. It is odd, since the `PATHS` option is supposed to have the lowest search priority, so it should find DSound in the compiler's paths instead of the manually-detected SDK path, unless the SDK version target that the compiler pre-selected doesn't have it, while the manually detected version target does.

I need to find out why exactly all this SDK searching seems necessary, and what the proper way is to handle whichever libraries need it manually specified.

**Comment by kcat** [comment]
This should be fixed now with commit 13698362f1726326ab60180b04a86df79b518614. It no longer explicitly looks for the WindowsSDK and uses the compiler's search paths (and only uses `DXSDK_DIR` for `dsound.h` if necessary).

**Comment by MikuAuahDark** [comment]
Nice, thanks. Just retested my local build also and no errors.
